CRR/ReplicationStatusUpdater.js

```diff
@@ -172,36 +175,54 @@ class ReplicationStatusUpdater {
                 //   codebase easier to maintain and upgrade, as opposed to having multiple branches or versions of
                 //   the code for different schema versions.
                 objMD = new ObjectMD(JSON.parse(mdRes.Body));
-                if (!this._objectShouldBeUpdated(objMD)) {
+                if (!this._objectShouldBeUpdated(objMD, storageClass)) {
                     skip = true;
                     return process.nextTick(next);
                 }
                 // Initialize replication info, if missing
                 // This is particularly important if the object was created before
                 // enabling replication on the bucket.
-                if (!objMD.getReplicationInfo()
-                    || !objMD.getReplicationSiteStatus(storageClass)) {
+                let replicationInfo = objMD.getReplicationInfo();
+                if (!replicationInfo || !replicationInfo.status) {
                     const { Rules, Role } = repConfig;
                     const destination = Rules[0].Destination.Bucket;
                     // set replication properties
                     const ops = objMD.getContentLength() === 0 ? ['METADATA']
                         : ['METADATA', 'DATA'];
-                    const backends = [{
-                        site: storageClass,
-                        status: 'PENDING',
-                        dataStoreVersionId: '',
-                    }];
-                    const replicationInfo = {
+                    replicationInfo = {
                         status: 'PENDING',
-                        backends,
                         content: ops,
+                        backends: [],
                         destination,
-                        storageClass,
+                        storageClass: '',
                         role: Role,
-                        storageType: this.storageType,
+                        storageType: '',
```

Comment:
what are these 2 fields (`storageClass` and `storageType`) used for?

Reply:
`storageClass` is what each location's replication queue processor uses to check if it should replicate an object or not. 
`storageType` is used by Cloudserver in Backbeat routes to do some pre-checks (check if versioning is supported on the backend and that the location is valid)

Reply:
that is how these fields are used... my question is really what information do they actually store, what these fields represent (for example, the storageClass of the object is already known, and stored in `.location[]` and `dataStoreName` ; and this cannot be the storageClass of the 'remote' object, since there may be multiple destinations...)

Reply:
the code which initializes these in cloudserver:
```
function _getReplicationInfo(rule, replicationConfig, content, operationType,
    objectMD, bucketMD) {
    const storageTypes = [];
    const backends = [];
    const storageClasses = _getStorageClasses(rule);
    if (!storageClasses) {
        return undefined;
    }
    storageClasses.forEach(storageClass => {
        const storageClassName =
              storageClass.endsWith(':preferred_read') ?
              storageClass.split(':')[0] : storageClass;
        const location = s3config.locationConstraints[storageClassName];
        if (location && replicationBackends[location.type]) {
            storageTypes.push(location.type);
        }
        backends.push(_getBackend(objectMD, storageClassName));
    });
    if (storageTypes.length > 0 && operationType) {
        content.push(operationType);
    }
    return {
        status: 'PENDING',
        backends,
        content,
        destination: replicationConfig.destination,
        storageClass: storageClasses.join(','),
        role: replicationConfig.role,
        storageType: storageTypes.join(','),
        isNFS: bucketMD.isNFS(),
    };
}
```

--> it seems these fields (along with `role`) are not multi-destination/rule aware...

Reply:
`storageClass` and `storageType` are multi-destination aware, they contain the list of all storage classes and storage types we are replicating to.
Example:
```
storageClass: "aws-location,azure-blob"
storageType: "aws_s3,azure"
```